aea/cli/upgrade.py

```diff
@@ -358,18 +250,27 @@ def check_upgrade_is_required(self) -> str:
 
     def remove_item(self) -> None:
         """Remove item from agent."""
-        remove_item(self.ctx, self.item_type, self.item_public_id)
-
-    def remove_dependencies_for_item(self) -> None:
-        """Remove all the dependecies not shared."""
-        for dep_package_id in self.deps_can_be_removed:
-            remove_item(
-                self.ctx, str(dep_package_id.package_type), dep_package_id.public_id
-            )
+        remove_item = RemoveItem(
+            self.ctx,
+            self.item_type,
+            self.item_public_id,
+            with_dependencies=True,
+            force=True,
+        )
+        remove_item.remove()
+        click.echo(f"Item { self.item_type} {self.item_public_id} removed!")
 
     def add_item(self) -> None:
         """Add new package version to agent."""
-        add_item(self.ctx, str(self.item_type), self.item_public_id)
+        current_item = get_item_public_id_by_author_name(
+            self.ctx.agent_config,
+            self.item_type,
+            self.item_public_id.author,
+            self.item_public_id.name,
+        )
+        if not current_item:
+            click.echo(f"Adding item { self.item_type} {self.item_public_id}.")
+            add_item(self.ctx, str(self.item_type), self.item_public_id)
```

Comment:
why has this changed?

Reply:
(also, remove unnecessary whitespace in click echo)

Reply:
going to remove after some checks. suppose it will be unused

Reply:
done